### ISSUE TITLE
Customizable text color

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#6638]
### Why this would be helpful

There are a few ways this could be useful:
- Making text less glaring (related: #5466)
- Increasing contrast to links (to make links easier to tell apart from normal text)

### My suggestion

Add the option to customize the text color in `dark-www` and maybe `editor-dark-mode`.

### Possible alternatives

The Customizable link style addon already helps with link differentiation and #5466 suggests making white text the slightest bit darker.

**Comment by WorldLanguages** [comment]
Is this a simple find-and-replace in the CSS files and creating a setting for it?

**Comment by Samq64** [comment]
I think only the addon.json needs to be updated since the text colour already has its own css variable.

**Comment by WorldLanguages** [comment]
@mxmou What do you think?

**Comment by mxmou** [comment]
One setting isn't enough - text in the navigation bar, for example, is white even in light mode. There could be one setting for text on light backgrounds and one for dark backgrounds. Another option is a to let the user choose a list of text colors and use the first one with good contrast. I think I prefer the second option - it might be harder to understand but it would allow more customization.

**Comment by DNin01** [comment]
> One setting isn't enough - text in the navigation bar, for example, is white even in light mode. There could be one setting for text on light backgrounds and one for dark backgrounds. Another option is a to let the user choose a list of text colors and use the first one with good contrast. I think I prefer the second option - it might be harder to understand but it would allow more customization.

Good point. What if the color for text on color (navbar, buttons) wasn't customizable so we didn't have to worry about that setting?

**Comment by mxmou** [comment]
> > One setting isn't enough - text in the navigation bar, for example, is white even in light mode. There could be one setting for text on light backgrounds and one for dark backgrounds. Another option is a to let the user choose a list of text colors and use the first one with good contrast. I think I prefer the second option - it might be harder to understand but it would allow more customization.
> 
> Good point. What if the color for text on color (navbar, buttons) wasn't customizable so we didn't have to worry about that setting?

That would still make certain customizations impossible - the 1.x theme for the editor, for example.

**Comment by WorldLanguages** [comment]
> There could be one setting for text on light backgrounds and one for dark backgrounds. Another option is a to let the user choose a list of text colors and use the first one with good contrast. I think I prefer the second option - it might be harder to understand but it would allow more customization.

The list of colors would probably be nice when creating presets. But for the average user it would be nicer to just display 1 or 2 settings. Maybe we can think of a way to support both cases at once.

**Comment by Secret-chest** [comment]
Maybe a Custom text colour boolean, which adds a table field with preferred text colour.

Another option is to make another colour picker UI with a smaller Text swatch next to it, which is Auto by default.

**Comment by ksdavidc** [comment]
Coming here from #6745, which was about the clone counter color, which involves both the text color and the color of the cat.svg, so perhaps off-topic. In addition, what I am about to suggest is a step or two farther out than what has been discussed so far, likely to be unpopular, probably a sinkhole to implement, not for newbies, and going to be shot down faster than you can say "delete clone". Also, I know jack-squat about how things are implemented, but am good at faking it, so take anything I say with a grain of salt, aspirin, or your sedative of choice. 

Here goes:

- when you are talking about the text color, I am not sure that affects the color of the clone counter text. The clone counter color seems to controllable via `.clone-container-container` or possibly `.clone-container` or `.clone-count`. 
- This does not affect the color of the cat.svg, which would have to be customized via something like 
```
.clone-icon {
 filter: invert(42%) sepia(93%) saturate(1352%) hue-rotate(87deg) brightness(119%) contrast(119%); 
}
```
- In wordpress, there is a custom css page with a text-area where you can put in your own css which it will apply to your template. Would it be possible to have such an area, say in the settings page as part of the editor colors, or other plugin (developer plugin?). 
- OF COURSE I am not suggesting this as a replacement for offering some kind of user-friendly alternative for adjusting the text color (I like the color picker idea, @Secret-chest), but for situations such as mine, this would not suffice. 
- I can guess the objections, and you can guess the replies.  whee.

**Comment by WorldLanguages** [comment]
> In wordpress, there is a custom css page with a text-area where you can put in your own css which it will apply to your template

There's an open source extension called Stylus that should help with that.
